core/runtime/runtime.go

```diff
@@ -0,0 +1,14 @@
+package runtime
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/stackitcloud/stackit-sdk-go/core/config"
+)
+
+// WithCaptureHTTPResponse adds the raw HTTP response retrieval annotation to the parent context.
+// The resp parameter will contain the raw HTTP response after the request has completed.
+func WithCaptureHTTPResponse(parent context.Context, resp **http.Response) context.Context {
+	return context.WithValue(parent, config.ContextHTTPResponse, resp)
```

Comment:
Maybe it also makes sense to move `ContextHTTPResponse` to the `runtime` package

Reply:
Then `runtime` would not need to have a dependency on `config`

Reply:
Does it break if we leave the current one during the deprecation period?
Then we would update the template to use the `runtime.ContextHTTPResponse` instead

Reply:
After than I think we could safely remove the deprecated method and var in `config`

Reply:
Then `config` imports `runtime` to use the new `ContextHTTPResponse`
That leaves us with `config` importing `runtime` instead of `runtime` importing `config`
Is that better?

Reply:
What I meant is that:
- `WithCaptureHTTPResponse` in `config`  uses the current `ContextHTTPResponse` from `config`
  - And these would be removed at the same time after the deprecation period
- While the new `WithCaptureHTTPResponse` in `runtime`  uses the new `ContextHTTPResponse` from `runtime`

Reply:
That breaks `WithCaptureHTTPResponse` from `config`

When the templates change, then they'll be checking `runtime.ContextHTTPResponse` which is different than `config.ContextHTTPResponse`. Doesn't matter that they have the same struct name and are built the same way, if one doesn't import from the other then they're different for Go.
(This is intentional, so that a package can use context keys with the same name without having to worry about collisions with keys from other packages)

Reply:
Ah you're right, because users can update their service version and keep using the old method from config